website/static/js/accountSettings.js

```diff
@@ -205,7 +211,8 @@ var UserProfileViewModel = oop.extend(ChangeMessageMixin, {
                         return;
                     }
                 }
-                this.changeMessage('Invalid Email.', 'text-danger');
+            }.bind(this)).fail(function(){
+                this.profile().emails.remove(email);
```

Comment:
Instead of popping it if the request fails, can we just make the push conditional on success? (see line 200)

What are the benefits of calling `this.profile().emails.push(email)`  outside of `this.client.update` succeeding? Just trying to make sure I understand the bug I filed. :)

That said, kudos to @chennan47 for jumping on this bug so fast!